### DISCUSSION TITLE
embedding

### DISCUSSION THREAD

**Discussion by BalajiVysya** [marp-team/marp#400] [Q&A] [answered]
tool is awesome, thanks for creating.

I was looking for good markdown plugin for VS code and this was the only tool i could find that fit my need.

I am facing one issue while embedding script tags in VS code. 

I tried the codes in the attached, but none of them worked. I am very new to markdown hope you can guide me.

---

[html.md](https://github.com/marp-team/marp/files/10336293/html.md)

**Comment by yhatt** [comment, marked as answer]
_We don't recommend using embedded scripts due to potential security risks (especially if you are not familiar with Markdown)._

Try following if you still want:

1. [Turn on raw HTML tags rendering from preference.](https://github.com/marp-team/marp-vscode#enable-html-in-marp-markdown-%EF%B8%8F)
2. Use [`<script>` HTML tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) like this. When opened exported HTML, the browser will load the script from URL in `src` attribute and run it.<br /><br />
   ```html
   <script src="https://.............."></script>
   ````
3. To run script with the slide preview provided by Marp for VS Code, you have to check following things:
   - [Trust the workspace that has editting Markdown.](https://code.visualstudio.com/docs/editor/workspace-trust)
   - If you could see the button "Some content has been disabled in this document" in the preview, disable [Markdown preview security](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview-security) by clicking that.

Again, _embedding script in Markdown is not recommended if not familiar with Marp and Markdown_ because the security was compromised by following these steps. Please do at your own risk.

**Comment by autogy** [comment]
thanks a lot for reverting back

sorry for the confusion, i was not actually looking for integrating a script.  

I am planning to use Marp as integrated documentation for software development.
I felt Marp is easy to use compared other documentation tools. 
the challenge i am facing in MARP is to embed github Premalink to show up. 
please see attached PNG "slide 3" right side is code snippet image from github flavored markdown i am trying to do the same in Marp


Example in my repo : https://github.com/BalajiVysya/markdown/blob/main/Demo.md

test reference File below: 
[Demo.md](https://github.com/marp-team/marp/files/10346663/Demo.md)


Slide 1
![slide 1](https://user-images.githubusercontent.com/121778896/210622373-f547685a-d73a-4e28-bd02-f28597335967.png)

Slide 2
![slide2](https://user-images.githubusercontent.com/121778896/210623820-aef52356-654a-407f-b389-e2155e166039.png)

slide 3
![slide3](https://user-images.githubusercontent.com/121778896/210624343-7629513d-4e41-4f4e-bff7-d1583f6365ac.png)

**Reply by yhatt** [reply]
The short answer is "you can't". Expanding code block from GitHub permalink URL is GitHub-specific feature.

In Markdown (CommonMark), using fenced code blocks is generally suited to embed codes in Markdown.
https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

````markdown
# Code block

```javascript
console.log('Hello, world!');
```
````

It's enough if using GitHub permalink URL is not important.

If you have to render a "live" snippet from the GitHub repository, using fenced code block may not suit. I don't know but a third-party service to generate a code block snippet as `<iframe>` compatible URL from GitHub permalink might be there in a world.

**Comment by BalajiVysya** [comment]
Thank you, for the clarification.
This helps.



________________________________
